mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -510,7 +510,12 @@ def as_system_method(self, args: T.Tuple[T.Optional[str]], kwargs: TYPE_kwargs)
     def as_link_whole_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> Dependency:
         if not isinstance(self.held_object, InternalDependency):
             raise InterpreterException('as_link_whole method is only supported on declare_dependency() objects')
-        new_dep = self.held_object.generate_link_whole_dependency()
+        try:
+            new_dep = self.held_object.generate_link_whole_dependency()
+        except MesonException as e:
+            raise MesonException(
+                e.args, self.current_node.filename, self.current_node.lineno,
+                self.current_node.colno)
```

Comment:
I don't think we need this, because interpreterbase's `evaluate_codeblock()` catches all exceptions, checks if it already has these properties, and if not, assigns filename/lineno/colno. Then it reraises the exception.